tests/functional/docs/test_shared_example_config.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import botocore.session
+from botocore.model import OperationNotFoundError
+from botocore.utils import parse_timestamp
+
+
+def test_lint_shared_example_configs():
+    session = botocore.session.Session()
+    loader = session.get_component('data_loader')
+    services = loader.list_available_services('examples-1')
+    for service in services:
+        service_model = session.get_service_model(service)
+        example_config = loader.load_service_model(
+            service, 'examples-1', service_model.api_version
+        )
+        examples = example_config.get("examples", {})
+        for operation, operation_examples in examples.items():
+            for example in operation_examples:
+                yield _lint_single_example, operation, example, service_model
+
+
+def _lint_single_example(operation_name, example_config, service_model):
+    # The operation should actually exist
+    assert_operation_exists(service_model, operation_name)
+    operation_model = service_model.operation_model(operation_name)
+    assert_valid_values(service_model.service_name, operation_model,
+                        example_config)
+
+
+def assert_valid_values(service_name, operation_model, example_config):
+    try:
+        example_input = example_config.get('input')
+        if example_input is not None:
+            _assert_valid_values(operation_model.input_shape, example_input)
+
+        example_output = example_config.get('output')
+        output_shape = operation_model.output_shape
+        if example_output is not None and output_shape is not None:
```

Comment:
Is there validation elsewhere if there's example output but not output shape?  Same question for input.  If not should that be a failure?